python/cudf/cudf/core/column/lists.py

```diff
@@ -76,7 +76,10 @@ def __sizeof__(self):
 
     @property
     def base_size(self):
-        return len(self.base_children[0]) - 1
+        # in some cases, libcudf will return an empty ListColumn with no
+        # indices; in these cases, we must manually set the base_size to 0 to
+        # avoid it being negative
+        return max(0, len(self.base_children[0]) - 1)
```

Comment:
Just noting this change; @shwina and I noticed that in some test cases, libcudf was returning empty a `ListColumn` with no indices (i.e. `self.base_children[0]` is just an empty `NumericalColumn`). Before, this would result in a `ListColumn` with a `base_size` of -1, which was breaking some test cases where this refactor required us to reconstruct the empty `ListColumn`.

This change ensures that `ListColumn.base_size` will always be at least 0.